src/fingerprinting/hooks.ts

```diff
@@ -0,0 +1,88 @@
+import { Fingerprint, FingerprintInjector } from 'fingerprint-injector';
+import { BrowserPool, PlaywrightPlugin, PuppeteerPlugin } from '..';
+import { BrowserController } from '../abstract-classes/browser-controller';
+import { LaunchContext } from '../launch-context';
+import { getGeneratorDefaultOptions } from './utils';
+
+// @ts-expect-error I need help how to fix this :/
+export const createFingerprintPreLaunchHook = (browserPool: BrowserPool) => {
```

Comment:
use generics for generic methods

```suggestion
export const createFingerprintPreLaunchHook = <T extends BrowserPoolOptions> (browserPool: BrowserPool<T>) => {
```

or if you it does not matter, you could use the `BrowserPoolOptions` directly (or as a default value for T):

```suggestion
export const createFingerprintPreLaunchHook = (browserPool: BrowserPool<BrowserPoolOptions>) => {
```

we should probably make it default value directly in the `BrowserPool` class, e.g.

```ts
export class BrowserPool<
    Options extends BrowserPoolOptions = BrowserPoolOptions,
    BrowserPlugins extends BrowserPlugin[] = InferBrowserPluginArray<Options['browserPlugins']>,
    // ...
```

that way your code will work without any changes

edit: but all of this will break use of `this` as the argument in the call site - just ts-expect-error that for now, i will need to redesign the `BrowserPool` type myself I guess, as this is close to unusable (in the TS SDK rewrite I am just feeding it as `BrowserPool<any, any, any, any, any, any>`, another way to deal with it for now :D)